tools/shared.py

```diff
@@ -1239,6 +1239,12 @@ def get(self, key):
     def __getitem__(self, key):
       return self.attrs[key]
 
+    @classmethod
+    def target_environment_may_be(self, environment):
+      SHELL_ENVIRONMENTS = ['spidermonkey', 'jsc', 'v8']
+      targeted_environments = self.attrs['ENVIRONMENT'].split(',')
```

Comment:
we don't intend for users to write `ENVIRONMENT=jsc` etc. - all we support there is `shell,web,worker,node`.

seems fine to me to keep the closure stuff in separate files, but it would be confusing to support `ENVIRONMENT=v8` etc. here, please remove it

Reply:
to be clearer, what I'm suggesting is that the closure method not call this with `v8`, `jsc`, etc., since it implies those are valid environments. Instead, how about if it calls this with `shell`, and in the closure method when that happens, it includes the `v8`, `jsc` etc. files?

Reply:
(as this is a detail of how the closure support is implemented, but those are not valid ENVIRONMENTs as that is defined in src/settings.js)

Reply:
Oh gotcha. Updated.